### PR TITLE
Wip test action

### PR DESCRIPTION
**Pull request by deusebio**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -71,20 +86,17 @@ def _write_github_output(urls_with_actions_dict: dict[str, str]) -> None:
...
-    else:
-        index_url = ""
-    github_output_path.write_text(
-        f"urls_with_actions={urls_with_actions}\nindex_url={index_url}\n",
-        encoding="utf-8",
-    )
+    output = ""
+
+    for key, urls_with_actions_dict in urls_with_actions_dicts.items():
+        output += f"{key}={_serialize_for_github(urls_with_actions_dict)}\n"
```
Comment (deusebio) [inline main.py:92]: use iter comprehension avoiding mutability of output

File: src/__init__.py [hunk 2]
```diff
@@ -88,66 +127,53 @@ def _run_reconcile(
...
+            "Cannot run any migration from Discourse as there is not discourse "
+            "link present in the metadata."
+        )
+        return {}
 
-    return {pr_link: ActionResult.SUCCESS}
+    logging.info(f"Tag exists: {clients.repository.tag_exists(DOCUMENTATION_TAG)}")
 
+    if not clients.repository.tag_exists(DOCUMENTATION_TAG):
+        with clients.repository.with_branch(DEFAULT_BRANCH) as repo:
```
Comment (deusebio) [inline src/__init__.py:157]: that should go into the repository

File: src/__init__.py [hunk 3]
```diff
@@ -88,66 +127,53 @@ def _run_reconcile(
...
+            main_hash = repo.current_commit
+        clients.repository.tag_commit(DOCUMENTATION_TAG, main_hash)
 
-def run(base_path: Path, discourse: Discourse, user_inputs: UserInputs) -> dict[str, str]:
-    """Interact with charmhub to upload documentation or migrate to local repository.
+    # Check difference with main
+    if not recreate_docs(clients, DOCUMENTATION_TAG):
```
Comment (deusebio) [inline src/__init__.py:162]: name  function `new_changes`

File: src/__init__.py [hunk 4]
```diff
@@ -88,66 +127,53 @@ def _run_reconcile(
...
+    pr_link = clients.repository.get_pull_request(DEFAULT_BRANCH_NAME)
 
-    Raises:
-        InputError: if no valid running mode is matched.
+    if pr_link is not None:
+        logging.info("upload-charm-documents pull request already open at %s", pr_link)
+        update_pull_request(clients.repository, DEFAULT_BRANCH_NAME)
+    else:
+        logging.info("PR not existing: creating a new one...")
+        pr_link = create_pull_request(clients.repository, DOCUMENTATION_TAG)
```
Comment (deusebio) [inline src/__init__.py:177]: place this in `pull_request.py`

File: src/repository.py [hunk 5]
```diff
@@ -459,18 +460,16 @@ def tag_commit(self, tag_name: str, commit_sha: str) -> None:
...
+            if self.tag_exists(tag_name):
+                logging.info("Removing tag %s", tag_name)
+                self._git_repo.git.tag("-d", tag_name)
+                self._git_repo.git.push("--delete", "origin", tag_name)
+
+            self._git_repo.git.tag(tag_name, commit_sha)
+            self._git_repo.git.push("origin", tag_name)
+
+        except GitCommandError as exc:
+            raise RepositoryClientError(f"Tagging commit failed. {exc=!r}") from exc
```
Comment (deusebio) [inline src/repository.py:472]: Let's make a tests where the tag tags a different commit than the one we are in

**Comment by jdkandersson** [comment]
Merged in another PR

**Pull request closed without merging**